cvat-core/src/frames.ts

```diff
@@ -555,7 +564,27 @@ export function getFramesMeta(type: 'job' | 'task', id: number, forceReload = fa
                     ...serialized,
                     deleted_frames: Object.fromEntries(serialized.deleted_frames.map((_frame) => [_frame, true])),
                 });
-                resolve(framesMetaData);
+                // When we get new framesMetaData from server there can be some unsaved data
+                // here we merge new meta data with cached one
```

Comment:
Consider extracting a function for this.

Reply:
Moved to separate function